sunpy/coordinates/_transformations.py

```diff
@@ -1071,7 +1071,7 @@ def gei_to_hme(geicoord, hmeframe):
     earth_object_int = geicoord.cartesian.transform(rot_matrix)
 
     # Find the Sun-object vector in the intermediate frame
-    sun_object_int = sun_earth_int + earth_object_int
+    sun_object_int = earth_object_int + sun_earth_int
```

Comment:
Why does this fix the bug?!

Reply:
Yup, abstraction makes this look silly.  What happens when you add two unit-bearing vectors is that the second addend is converted to the unit of the first addend, then added, so the output will always have the unit of the first addend.  In this case, `earth_object_int` has the initial units, and `sun_earth_int` has AU units, so we want `earth_object_int` first.  We of course could instead explicitly transform to the unit afterwards, but it's unnecessary as long as we're intentional about addend order.

Reply:
That makes sense, thanks for the explanation.
I wasn't sure what the type of either earth_object_int and sun_earth_int.

Reply:
They are both `CartesianRepresentation`, with the relevant behavior actually coming from `Quantity`

Reply:
I think what @samaloney said in chat is valid, we should add a comment here.
```suggestion
    # Add sun_earth_int to earth_object_int to maintain the units of earth_object_int in the result.
    sun_object_int = earth_object_int + sun_earth_int
```

Reply:
Added a comment